src/rules/no-restricted-paths.js

```diff
@@ -56,78 +56,125 @@ module.exports = {
+      }
 
-        if (hasValidExceptions) {
-          const exceptionsMm = zoneExcept.map((except) => new Minimatch(except));
-          isPathException = (absoluteImportPath) => exceptionsMm.some((mm) => mm.match(absoluteImportPath));
-        }
+      const reportInvalidException = reportInvalidExceptionGlob;
 
-        reportInvalidException = reportInvalidExceptionGlob;
-      } else {
-        isPathRestricted = (absoluteImportPath) => containsPath(absoluteImportPath, absoluteFrom);
+      return {
+        isPathRestricted,
+        hasValidExceptions,
+        isPathException,
+        reportInvalidException,
+      };
+    }
 
-        const absoluteExceptionPaths = zoneExcept
-          .map((exceptionPath) => path.resolve(absoluteFrom, exceptionPath));
-        hasValidExceptions = absoluteExceptionPaths
-          .every((absoluteExceptionPath) => isValidExceptionPath(absoluteFrom, absoluteExceptionPath));
+    function computeAbsolutePathValidator(absoluteFrom, zoneExcept) {
+      let isPathException;
 
-        if (hasValidExceptions) {
-          isPathException = (absoluteImportPath) => absoluteExceptionPaths.some(
-            (absoluteExceptionPath) => containsPath(absoluteImportPath, absoluteExceptionPath),
-          );
-        }
+      const isPathRestricted = (absoluteImportPath) => containsPath(absoluteImportPath, absoluteFrom);
 
-        reportInvalidException = reportInvalidExceptionPath;
+      const absoluteExceptionPaths = zoneExcept
+        .map((exceptionPath) => path.resolve(absoluteFrom, exceptionPath));
+      const hasValidExceptions = absoluteExceptionPaths
+        .every((absoluteExceptionPath) => isValidExceptionPath(absoluteFrom, absoluteExceptionPath));
+
+      if (hasValidExceptions) {
+        isPathException = (absoluteImportPath) => absoluteExceptionPaths.some(
+          (absoluteExceptionPath) => containsPath(absoluteImportPath, absoluteExceptionPath),
+        );
       }
 
+      const reportInvalidException = reportInvalidExceptionPath;
+
       return {
         isPathRestricted,
         hasValidExceptions,
         isPathException,
         reportInvalidException,
       };
+    }
+
+    const makePathValidators = (zoneFrom, zoneExcept = []) => {
+      const allZoneFrom = zoneFrom instanceof Array ? zoneFrom : [zoneFrom];
+      const areGlobPatterns = allZoneFrom.map(isGlob);
+
+      if (areBothGlobPatternAndAbsolutePath(areGlobPatterns)) {
+        return [computeMixedGlobAndAbsolutePathValidator()];
+      }
+
+      const isGlobPattern = areGlobPatterns.every((isGlob) => isGlob);
+
+      return allZoneFrom.map(singleZoneFrom => {
+        const absoluteFrom = path.resolve(basePath, singleZoneFrom);
+
+        if (isGlobPattern) {
+          return computeGlobPatternPathValidator(absoluteFrom, zoneExcept);
+        } else {
+          return computeAbsolutePathValidator(absoluteFrom, zoneExcept);
+        }
```

Comment:
```suggestion
        }
        return computeAbsolutePathValidator(absoluteFrom, zoneExcept);
```

Reply:
@ljharb seems like a formatting preference, ok to align on yours 👍🏽 https://github.com/import-js/eslint-plugin-import/pull/2466/commits/6b4ea9f5f09db7e440883a87e4627c349b6f7f4e